segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+        # can't be centered up
+        if coordinate_pair[1] < (math.ceil(box_size[1]/2) - 1):
+            transformed_y = math.ceil(box_size[1] / 2) - 1
+
+        # can't be centered right
+        if (coordinate_pair[0] + (math.ceil(box_size[0]/2))) > video_shape[0] - 1:
+            transformed_x = video_shape[0] - math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered down
+        if (coordinate_pair[1] + (math.ceil(box_size[1]/2))) > video_shape[1] - 1:
+            transformed_y = video_shape[1] - math.ceil(box_size[1] / 2) - 1
+
+        return transformed_x, transformed_y
+
+    else:
+        raise ValueError('Box is larger or equal in size in one or more'
+                         'dimensions to original video shape. Provide a '
+                         'box size smaller than original video.')
+
+
+def get_centered_coordinate_box(coordinate_pair: Tuple[int, int],
+                                box_size: Tuple[int, int],
+                                video_array: np.ndarray):
+    """
+    Function to get video subset centered around coordinate pair with
+    size specified by box size tuple. Function takes a subset of each frame
+    and stacks together to get final video.
+    Args:
+        coordinate_pair: the coordinate on which to center the subset
+        box_size: the size of the subset box
+        video_array: the video to take the subset from
+
+    Returns:
+
+    """
+    transformed_coordinates = get_transformed_center(coordinate_pair, box_size,
+                                                     video_array.shape)
+    left_column = transformed_coordinates[0] - math.ceil(box_size[0] / 2)
+    right_column = transformed_coordinates[0] + math.ceil(box_size[0] / 2)
+    up_row = transformed_coordinates[1] - math.ceil(box_size[1] / 2)
+    down_row = transformed_coordinates[1] + math.ceil(box_size[1] / 2)
+    transformed_stack = []
+    for frame in video_array:
+        transformed_frame = frame[up_row:down_row, left_column:right_column]
+        transformed_stack.append(transformed_frame)
+    transformed_video = np.stack(transformed_stack)
+    return transformed_video
+
+
+def generate_max_ave_proj_image(video: np.ndarray, projection_type: str= 'average'):
+    """
+    Returns a maximum projection or an average projection of a video in
+    numpy array format
+    Args:
+        video: The video to generate the projection
+        projection_type: maximum or average, what type of projection
+
+    Returns: A numpy array generated with the specified strategy
+
+    """
+    if projection_type == 'average':
+        return np.mean(video)
+    elif projection_type == 'maximum':
+        return np.max(video, axis=0)
+
+
+def normalize_video(video: np.ndarray):
+    """
+    Function to normalize video by its global max and maximum 8 bit value
+    Args:
+        video: Video to be normalized
+    Returns:
+        norm_frames: normalized video
+    """
+    global_max = video.max()
+    norm_frames = []
```

Comment:
Similar to above comment about allocation. For this one just create an empty array and insert the appropriate frames.